From 56f5af15bc4796cf903ef5e9f82ed8514190e6bf Mon Sep 17 00:00:00 2001
From: Adrian Ratiu <adrian.ratiu@collabora.com>
Date: Tue, 21 May 2024 10:38:37 +0300
Subject: [PATCH 2/3] meson.build: reduce version duplication

Gentoo dev Matt Turner asked us to reduce version duplication
in the meson.build files, to decrease the risks of getting
the version wrong like in the past by forgetting to bump the
minor version everywhere the version number is hardcoded.

Upstream-Status: In progress [https://github.com/libtypec/libtypec/pull/1]

Signed-off-by: Adrian Ratiu <adrian.ratiu@collabora.com>
---
 meson.build       | 11 ++++++-----
 utils/meson.build |  9 +++++----
 2 files changed, 11 insertions(+), 9 deletions(-)

diff --git a/meson.build b/meson.build
index 5512f77..7a7c28a 100644
--- a/meson.build
+++ b/meson.build
@@ -5,9 +5,10 @@ default_options : [
 	'warning_level=0'])
 
 conf_data = configuration_data()
-conf_data.set('libtypec_VERSION_MAJOR', '0')
-conf_data.set('libtypec_VERSION_MINOR', '5')
-conf_data.set('libtypec_VERSION_PATCH', '1')
+split = meson.project_version().split('.')
+conf_data.set('libtypec_VERSION_MAJOR', split[0])
+conf_data.set('libtypec_VERSION_MINOR', split[1])
+conf_data.set('libtypec_VERSION_PATCH', split[2])
 
 libudev_dep = dependency('libudev', required: true)
 
@@ -17,7 +18,7 @@ library('typec',
 	'libtypec.c',
 	'libtypec_sysfs_ops.c',
 	'libtypec_dbgfs_ops.c',
-	version : '0.5.1',
+	version : meson.project_version(),
 	soversion : '5',
 	dependencies: libudev_dep,
-)
\ No newline at end of file
+)
diff --git a/utils/meson.build b/utils/meson.build
index 8519541..9bd2245 100644
--- a/utils/meson.build
+++ b/utils/meson.build
@@ -1,13 +1,14 @@
 project('libtypec_utils','c',
 license: 'MIT',
+version: '0.5.1',
 default_options : [
 	'warning_level=0'])
 
 conf_data = configuration_data()
-conf_data.set('libtypec_utils_VERSION_MAJOR', '0')
-conf_data.set('libtypec_utils_VERSION_MINOR', '5')
-conf_data.set('libtypec_utils_VERSION_PATCH', '1')
-
+split = meson.project_version().split('.')
+conf_data.set('libtypec_utils_VERSION_MAJOR', split[0])
+conf_data.set('libtypec_utils_VERSION_MINOR', split[1])
+conf_data.set('libtypec_utils_VERSION_PATCH', split[2])
 
 configure_file(input : 'libtypec_utils_config.h.in', output : 'libtypec_utils_config.h', configuration : conf_data)
 
-- 
2.44.1

